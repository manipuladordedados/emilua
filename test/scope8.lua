-- Cleanup handlers that suspend

spawn(function()
    scope_cleanup_push(function()
        print('foo1')
        this_fiber.yield()
        print('foo2')
    end)
    error('foo3')
end):detach()

spawn(function()
    scope_cleanup_push(function()
        print('bar1')
        this_fiber.yield()
        print('bar2')
    end)
    error('bar3')
end):detach()
