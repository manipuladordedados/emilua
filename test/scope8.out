foo1
bar1
foo2
Fiber 0x1 panicked: 'input.lua:9: foo3'
stack traceback:
	input.lua:9: in function <input.lua:3>
	[C]: in function ''
	[string "?"]: in function <[string "?"]:0>
bar2
Fiber 0x2 panicked: 'input.lua:18: bar3'
stack traceback:
	input.lua:18: in function <input.lua:12>
	[C]: in function ''
	[string "?"]: in function <[string "?"]:0>
